aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -181,47 +181,47 @@ static void norm_kernel_tensor_iterator_impl(
 
 
   if (val == 0) {
-    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(kHalf, iter.dtype(), "norm_cpu", [&] {
+    AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "norm_cpu", [&] {
```

Comment:
this enables bfloat16 for cpu norm, but not for gpu norm (also, intermediate accumulation is still bfloat16, which is very imprecise, but that's a problem for another day).